lib/hedwig/transports/tcp.ex

```diff
@@ -163,6 +164,14 @@ defmodule Hedwig.Transports.TCP do
     {:noreply, new_state}
   end
 
+  @doc """
+  Gaurd for closing idle stream. xmpp.org XEP-0190
+  """
+  defp handle_data(_socket, data, state) when data == "</stream:stream>" do
+    Logger.warn fn -> "Incoming close idle stream: #{inspect data}" end
+    Client.handle_info({:stop, "Idle stream shutdown."}, Client.get(state.client))
```

Comment:
`handle_info` is an OTP callback that is invoked by send a process a message. It is not meant to be called directly.